src/Data/Text/IO/Utf8.hs

```diff
@@ -0,0 +1,44 @@
+-- |
+-- Module      : Data.Text.IO.Utf8
+-- Copyright   : (c) 2009, 2010 Bryan O'Sullivan,
+--               (c) 2009 Simon Marlow
+-- License     : BSD-style
+-- Maintainer  : bos@serpentine.com
+-- Portability : GHC
+--
+-- Efficient UTF-8 support for text I\/O.
+module Data.Text.IO.Utf8
+    (
+      readFile
+    , writeFile
+    , appendFile
+    ) where
+
+import Prelude hiding (readFile, writeFile, appendFile)
+import Control.Exception (evaluate)
+import Control.Monad ((<=<))
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as B
+import Data.Text (Text)
+import Data.Text.Encoding (decodeUtf8, encodeUtf8)
+
+decodeUtf8IO :: ByteString -> IO Text
+decodeUtf8IO = evaluate . decodeUtf8
```

Comment:
This use of `evaluate` makes sense to me - if `decodeUtf8` throws an error, then we want that to blow up at the `readFile` call-site and not the evaluation site of the returned `Text`.

I feel like this signature would benefit from a `HasCallStack` constraint, since it can throw (along with `readFile` below).

Reply:
I don't think adding it will do anything, because `decodeUtf8` doesn't have `HasCallStack`